### PR TITLE
Heat solver using scalar framework - Part 1

### PR DESCRIPTION
**Pull request by pcarruscag**

### CODE REVIEW

**Comment by bigfooted** [comment]
Nice work, looks good as always, so what is left for the WIP?

File: SU2_CFD/src/solvers/CHeatSolver.cpp [hunk 1]
```diff
@@ -26,13 +26,16 @@
  */
 
 #include "../../include/solvers/CHeatSolver.hpp"
+#include <cstddef>
 #include "../../../Common/include/toolboxes/geometry_toolbox.hpp"
+#include "../../include/solvers/CScalarSolver.inl"
 
-CHeatSolver::CHeatSolver(CGeometry *geometry, CConfig *config, unsigned short iMesh) : CSolver(),
-             flow(config->GetFluidProblem()), heat_equation(config->GetHeatProblem()) {
+/*--- Explicit instantiation of the parent class of CTurbSolver. ---*/
```
Comment (WallyMaier) [inline SU2_CFD/src/solvers/CHeatSolver.cpp:33]: Should this comment have CScalarSolver?

Comment (pcarruscag) [reply SU2_CFD/src/solvers/CHeatSolver.cpp:33]: yes

Comment (pcarruscag) [inline SU2_CFD/src/solvers/CHeatSolver.cpp:33]:
```suggestion
/*--- Explicit instantiation of the parent class of CHeatSolver. ---*/
```

**Review by WallyMaier — APPROVED** [review]
LGTM!

**Pull request merged**